pkg/commands/git_commands/stash.go

```diff
@@ -49,6 +49,10 @@ func (self *StashCommands) ShowStashEntryCmdObj(index int) oscommands.ICmdObj {
 	return self.cmd.New(cmdStr).DontLog()
 }
 
+func (self *StashCommands) SaveUnstagedChanges(message string) error {
+	return self.cmd.New("git stash --keep-index").Run()
```

Comment:
Although this does exactly want the linked issue wanted, I've just tested this and it actually doesn't only stash the unstaged changes. Instead, it stashes both staged and unstaged, but leaves the staged changes in the index. This stack overflow goes into some detail: https://stackoverflow.com/questions/20028507/git-stash-uncached-how-to-put-away-all-unstaged-changes

So I'm thinking we just change the label on the menu item to 'stash all changes and keep index'. I'd also give it a separate keybinding like 'i' for 'index' (can't use 'k' due to vim keybindings). That means later on we can add a keybinding for actually only stashing unstaged changes. What do you think?